scripts/process-reboot-cause

```diff
@@ -69,6 +71,84 @@ def read_reboot_cause_files_and_save_state_db():
                 x = TIME_SORTED_FULL_REBOOT_FILE_LIST[i]
                 os.remove(x)
 
+def get_dpus():
+    """Retrieve the list of DPUs from the platform.json file."""
+    dpus = []
+
+    try:
+        # Get platform information
+        platform_info = device_info.get_platform_info()
+        platform = platform_info.get('platform')
+        if not platform:
+            raise KeyError("Platform key missing from platform_info")
+
+        # Path to platform.json
+        platform_path = os.path.join("/usr/share/sonic/device", platform, "platform.json")
+        sonic_logger.log_info(f"Reading platform JSON from: {platform_path}")
+
+        with open(platform_path, 'r') as f:
+            platform_data = json.load(f)
+
+        dpus = platform_data.get('DPUS', [])
+        if not dpus:
+            sonic_logger.log_warning("No DPUS found in platform.json")
+
+    except Exception as e:
+        sonic_logger.log_error(f"Error retrieving DPUs: {e}")
+
+    return dpus
+
+def get_sorted_reboot_cause_files(dpu_history_path):
+    """Retrieve and sort the reboot cause files for a specific DPU."""
+    try:
+        # Assuming you have a way to list the files in the directory
+        files = os.listdir(dpu_history_path)
+        # Filter and sort the files based on your criteria (e.g., by modification time)
+        sorted_files = sorted(
+            [os.path.join(dpu_history_path, f) for f in files if f.endswith('.json')],
+            key=os.path.getmtime,  # Sort by modification time
+            reverse=True  # Most recent first
+        )
+        return sorted_files
+    except Exception as e:
+        sonic_logger.log_error(f"Error retrieving reboot cause files for {dpu_history_path}: {e}")
+        return []
+
+
+def read_dpu_reboot_cause_files_and_save_chassis_state_db():
+    """Retrieve reboot cause from history files and save them to StateDB."""
+    try:
+        # Get the DPUs from the platform configuration
+        dpus = get_dpus()
+        history_dir = '/host/reboot-cause/module'
+
+        for dpu in dpus:
+            dpu_history_path = os.path.join(history_dir, dpu)
+
+            # Get sorted reboot cause files for the DPU
+            reboot_files = get_sorted_reboot_cause_files(os.path.join(dpu_history_path, "history"))
+
+            for reboot_file in reboot_files:
```

Comment:
@rameshraghupathy How do we handle a case where NPU comes late so that DPU to NPU mid plane is not UP by the time this process starts?

Reply:
@prgeor As shown in the HLD the NPU-chassisd will fetch the reboot-cause from the DPU and persist it.